python_modules/dagster/dagster/core/definitions/sensor_definition.py

```diff
@@ -224,12 +228,16 @@ def __init__(
                     ),
                 )
             ]
-        elif job:
-            targets = [DirectTarget(job)]
+            self._jobs = None
```

Comment:
Similar question as in ScheduleDefinition - should the pipeline name be incorporated into the jobs property in some way even though it's old api?

Reply:
I don't think so. Hard erroring is fine imo